plugins/examples/stable-analysis/src/main/java/org/elasticsearch/example/analysis/CustomAnalyzerFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.example.analysis;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.elasticsearch.example.analysis.lucene.CharSkippingTokenizer;
+import org.elasticsearch.example.analysis.lucene.ReplaceCharToNumber;
+import org.elasticsearch.example.analysis.lucene.SkipStartingWithDigitTokenFilter;
+import org.elasticsearch.plugin.api.NamedComponent;
+import org.elasticsearch.plugin.api.Inject;
+
+import java.util.List;
+
+@NamedComponent( "example_analyzer_factory")
+public class CustomAnalyzerFactory implements org.elasticsearch.plugin.analysis.api.AnalyzerFactory {
+    private final ExampleAnalysisSettings settings;
+
+    @Inject
+    public CustomAnalyzerFactory(ExampleAnalysisSettings settings) {
+        this.settings = settings;
+    }
+
+    @Override
+    public Analyzer create() {
+        return new CustomAnalyzer(settings);
+    }
+
+    static class CustomAnalyzer extends Analyzer {
+
+        private final ExampleAnalysisSettings settings;
+
+        public CustomAnalyzer(ExampleAnalysisSettings settings) {
+            this.settings = settings;
+        }
+
+        @Override
+        protected TokenStreamComponents createComponents(String fieldName) {
+            var tokenizerListOfChars = settings.singleCharsToSkipInTokenizer().isEmpty() ? List.of("_") : settings.singleCharsToSkipInTokenizer();
```

Comment:
does ListSetting not have a definable default?

Reply:
it does not. it returns an empty list.
but this is good point, perhaps we should have a default. I will create an issue to follow up. this might be problem when defining a default for a list of custom types.